tests/Doctrine/Tests/ORM/Query/QueryTest.php

```diff
@@ -303,4 +304,65 @@ public function testSelectFromSubquery()
         $this->expectExceptionMessage('Subquery');
         $query->getSQL();
     }
+
+    /**
+     * @group #6699
+     */
+    public function testGetParameterTypeJuggling() : void
+    {
+        $query = $this->_em->createQuery('select u from ' . CmsUser::class . ' u where u.id = ?0');
+
+        $query->setParameter(0, 0);
+
+        $this->assertSame(0, $query->getParameter(0)->getValue());
+        $this->assertSame(0, $query->getParameter('0')->getValue());
+    }
+
+    /**
+     * @group #6699
```

Comment:
We're using the ticket id without the `#` everywhere else, we should keep it consistent here too